.travis.yml

```diff
@@ -1,48 +1,95 @@
 env:
   global:
-  - NODE_VERSION=8.9.0
-  - EMULATOR_API_LEVEL=19
-  - ANDROID_VERSION=25
-  - ANDROID_BUILD_TOOLS_VERSION=25.0.2
-  - ANDROID_ABI=armeabi-v7a
-  - APPIUM_VERSION=1.7.0
-  - EMULATOR_NAME=test
-language: android
-jdk:
-- oraclejdk8
+    - NODE_VERSION=8.9.0
+    - EMULATOR_API_LEVEL=22
+    - ANDROID_VERSION=25
+    - ANDROID_BUILD_TOOLS_VERSION=25.0.2
+    - ANDROID_ABI=armeabi-v7a
+    - APPIUM_VERSION=1.7.0
+    - EMULATOR_NAME=test
+    - ANDROID_TAG=google_apis
+
+matrix:
+  include:
+    - stage: "Lint"
+      language: node_js
+      os: linux
+      node_js: "6"
+      script: 
+        - cd nativescript-angular
+        - npm install
+        - npm run tslint
+    - stage: "Build"
+      os: osx
+      env:
+        - Build="iOS"
+      osx_image: xcode8.3
+      language: node_js
+      node_js: "8"
+      jdk: oraclejdk8
+      before_install: 
+        - cd nativescript-angular
+        - npm install
+        - npm run ngc
+        - cd ..
+      script: 
+        - tns build ios
+        - cd tests
+        - npm install
+    - language: android
+      os: linux
+      env:
+        - Build="Android"
+      jdk: oraclejdk8
+      before_install: 
+        - echo no | android create avd --force -n $EMULATOR_NAME -t "android-"$EMULATOR_API_LEVEL --abi $ANDROID_ABI --tag $ANDROID_TAG -c 12M
+        - QEMU_AUDIO_DRV=none emulator -avd $EMULATOR_NAME -no-window &
+        - nvm install $NODE_VERSION
+        - cd nativescript-angular
+        - npm install
+        - npm run ngc
+        - cd ..
+        - android-wait-for-emulator
+        - adb shell input keyevent 82 &
+      script: 
+        - tns build android
```

Comment:
I think this should be the last step (after navigating to tests and running npm install)

Reply:
@sis0k0  the adb shell or the build?

Reply:
The build

Reply:
ping @hypery2k  ^